accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,337 @@
+
+# Supported installation types
+
+We plan to (eventually) support the following installation types.  Some of them may not be supported in .NET 6 if we don't have workloads in .NET 6 that are supported on the corresponding platforms.
+
+## .NET SDK managed installation
+
+In this implementation of the IAL, the .NET SDK itself manages workload installations directly by writing files to the dotnet folder.  This is generally for copies of the .NET SDK which were not installed via an OS installer, for example those that are installed via the dotnet-install scripts, or via unzipping archives of the .NET SDK.
+
+Other terms we have at times used for this installation type: xcopy, universal installer, files on disk, NuGet installer
+
+## Visual Studio managed installation
+
+This is an installation of the .NET SDK installed and managed by the Visual Studio installer.  The .NET SDK workloads will have corresponding Visual Studio workloads or components, which will install workload packs via Windows MSIs.  Note that the Visual Studio workloads or components should also include the necessary Visual Studio components (designers, project system, etc) to support the workload, in addition to the .NET SDK workload packs.
+
+## Standalone Windows installation
+
+This is an installation of the .NET SDK installed on Windows via the .NET SDK installer bundle, which is the .exe which can be downloaded on the .NET SDK download pages.  The bundle installs the same MSIs for the .NET SDK as Visual Studio does, and the workload installer will likewise install the same MSIs for workload packs as the Visual Studio installer.
+
+The same dotnet folder may have different versions of the SDK installed by both the Visual Studio installer and the Standalone installer, which may share the same versions of some components.  For the .NET SDK, the underlying MSIs are ref-counted so that they are uninstalled when there are no more installations that depend on them.  We will need to make sure that workload installation and management also works well when there are different install types coexisting.
+
+## Package manager installation
+
+This is a copy of the .NET SDK installed as a package from a package manager such as APT or RPM.  In this case, the workloads and / or workload packs will also be represented as packages which will be installed via the package manager.
+
+# Choosing the installer abstraction
+
+Before any installation operation is started, the .NET SDK will need to choose which installer abstraction will be used and instantiate it.  To do so, it will look for marker files in the `<DOTNET ROOT>/sdk-manifests/<SdkFeatureBand>/.installertype/` folder.  If a `visualstudio` file is present in that folder, the Visual Studio installer abtraction will be used.  Otherwise, if an `msi` file is present, the standalone Windows installation abstraction will be used.  Otherwise, the .NET SDK managed installer abstraction will be used.
```

Comment:
Any reason to use hidden files/directories (`.installertype`)? Some programs like [rkhunter](http://rkhunter.sourceforge.net/) will flag these as suspicious.

Reply:
@mhutch had similar feedback, and we are going to change this to use different folders (under a `metadata` folder in the root `dotnet` folder).